src/common/utils/setup-helpers.js

```diff
@@ -357,7 +357,7 @@ function checkNativeHlsSupport(options: KPOptionsObject): void {
  * @returns {void}
  */
 function checkNativeTextTracksSupport(options: KPOptionsObject): void {
-  if ((Env.isMacOS && Env.isSafari) || Env.isIOS || (options.text && options.text.useShakaTextTrackDisplay)) {
+  if ((Env.isMacOS && Env.isSafari) || Env.isIOS) {
```

Comment:
you can use getPropertyPath
what happens if this is true but the selected source is hls instead of dash ?